src/main/java/io/github/wimdeblauwe/hsbt/mvc/HxTrigger.java

```diff
@@ -0,0 +1,12 @@
+package io.github.wimdeblauwe.hsbt.mvc;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface HxTrigger {
```

Comment:
It would be nice to have `afterSettle` and `afterSwap` in the same annotation instead of as separate annotations. I think `value` aligns with the base one though. 

Also, we could have a single annotation called `@HxResponse` with `trigger`, `triggerAfterSettle`, `triggerAfterSwap` (and the other [potential response headers](https://htmx.org/reference/#response_headers) )

Reply:
Interesting remark. I had a look at [django-htmx and they call it `trigger_client_event`](https://django-htmx.readthedocs.io/en/latest/http.html#django_htmx.http.trigger_client_event) with 3 options for the `after` variable. We could expand the `@HxTrigger` annotation to also have this kind of parameter with an enum representing the 3 options? For the other response headers, I would be more in favor of adding `@HxRefresh`, `@HxPush`, etc..

Reply:
I updated the PR to showcase this idea. How do you feel about it?

Reply:
'Policy' isn't the first thing that comes to mind 'lifecycle' might be closer.  I also agree that having different annotations for each header is a good idea. However by having a general `@HxResponse` all the other headers can be implemented using the inheritance mechanism.

Reply:
Worth mentioning, this approach stops a user from setting all 3 headers at once (not sure why that would be needed). I haven't wrapped my head around why after swap/settle are used yet though, so my experience is limited.

Reply:
> 'Policy' isn't the first thing that comes to mind 'lifecycle' might be closer. I also agree that having different annotations for each header is a good idea. 

I'll rename to `HxTriggerLifecycle`

Reply:
> Worth mentioning, this approach stops a user from setting all 3 headers at once (not sure why that would be needed).

Also no idea. Let's leave it like this until somebody comes up with a usecase where this would be needed.